docs/guides/src/main/server/configuring-database.adoc

```diff
@@ -0,0 +1,126 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/options.adoc" as opts>
+
+<@tmpl.guide
+    title="Configuring a database"
+    summary="An overview about how to configure relational databases"
+    includedOptions="db db-*">
+
+In this guide, you are going to understand how to configure the server to store data using different relational databases. You should also learn
+what are the databases supported by the server and how to proceed when using a unsupported database version.
+
+== Querying the list of supported databases
+
+The server has built-in support for different databases. You should be able to query the available databases by looking at the expected values
+for the `db` configuration option.
+
+The list of supported databases and their corresponding versions are:
+
+|===
+|Database | Version
+
+|mariadb| 10.3
+|mssql| 2016
+|mysql| 8
+|oracle| 12c
+|postgres| 10
+|===
+
+[NOTE]
+The server is opinionated about the databases and their respective versions. If you need support for a different database, there are some
+advanced configuration options to override the default configuration (e.g.: JDBC driver, Hibernate ORM Dialect, disable XA, etc). In this case, please help us by creating an issue to discuss
+the possibility to support additional databases.
```

Comment:
I would remove this section, as we will not support any relational database vendor out there, now or in the future. Some more thoughts on this:

* We should update our testing more regularly with more recent versions of the databases, but if we just change it to `Tested version` it leaves it open for folks to open an issue against newer versions if they don't work. We don't need to specify that in the docs though (as otherwise we'd have mentions like this everywhere)
* Cloud vendor specific versions like AWS RDS, we need to prioritise which we want to test/support, and when we do should cover this directly without extra config. I could be open for a section at the end mentioning some instructions on things to configure/tune to try to get it to work with various variants, but I'd leave that for later, and only if we have some instructions to provide